### PR TITLE
Update packages for 7.1.0

### PR DESCRIPTION
**Pull request by thisac**
Update packages and subprojects to latest releases.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -48,4 +48,4 @@ python_requires = >=3.8
 
 [options.extras_require]
 all =
-    dwave-gate==0.3.1
+    dwave-gate==0.3.2
```
Comment (arcondello) [inline setup.cfg:51]: Did we want to promote `dwave-gate` at the same time?

**Review by arcondello — APPROVED** [review]

File: setup.cfg (hunk 1 above)
Comment (thisac) [reply setup.cfg:51]: We could, should we? Unless anyone thinks otherwise.

Comment (arcondello) [reply setup.cfg:51]: :shrug: No preference either way.

**Comment by arcondello** [comment]
Doctest failures look like they are mostly from NumPy object printing changes in NumPy 2.0. Maybe makes sense to bound `Numpy<2.0` in the doctests for now?

File: setup.cfg (hunk 1 above)
Comment (thisac) [reply setup.cfg:51]: Alright, let's promote it now then. I'll think about how to best structure other extras (specifically the QIR dependency in dwave-gate) later.

**Comment by arcondello** [comment]
Oh, also, I think we should add 
```python
autodoc_type_aliases = {
    'numpy.typing.ArrayLike': 'numpy.typing.ArrayLike',
}
```
 to `conf.py` for slightly more readable docs in some places. There are other places we could/should add aliases as well, but that one will definitely help with `dwave-optimization` generator docs.

File: setup.cfg [hunk 2]
```diff
@@ -45,7 +46,3 @@ install_requires =
     # pyqubo==1.4.0
 packages = dwaveoceansdk
 python_requires = >=3.8
-
-[options.extras_require]
-all =
```
Comment (arcondello) [inline setup.cfg:50]: I would leave `all` in with nothing. Because otherwise any `pip install dwave-ocean-sdk[all]` out there will break.

Comment (thisac) [reply setup.cfg:50]: Ah, good point.

File: setup.cfg [hunk 3]
```diff
@@ -26,26 +26,23 @@ description = Software development kit for open source D-Wave tools
...
     dwave-inspector==0.5.0.post0
     dwave-neal==0.6.0
     dwave-networkx==0.8.15
-    dwave-optimization==0.1.0
-    dwave-preprocessing==0.6.5
-    dwave-samplers==1.2.0
+    dwave-optimization==0.2.0
+    dwave-preprocessing==0.6.6
+    dwave-samplers==1.3.0
     dwave-system==1.25.0
```
Comment (randomir) [inline setup.cfg:40]: We could also release new dwave-system, but since there're only minor test and docs improvements, we can skip it this time as well.

**Review by randomir — APPROVED** [review]
lgtm

File: setup.cfg (hunk 2 above)
Comment (thisac) [reply setup.cfg:50]: Seems like only a warning is produced when calling an extra that doesn't exist though, so probably no big harm in not having it. It's there as an empty extra now anyway.

**Comment by thisac** [comment]
@arcondello I don't see what `autodoc_type_aliases = {'numpy.typing.ArrayLike': 'numpy.typing.ArrayLike'}` would do. Shouldn't the key be a _different_ alias than the value?

**Comment by arcondello** [comment]
Docs on this branch
image

Docs with the alias
image


This is basically saying "replace any instance of `numpy.typing.ArrayLike` with the literal string `numpy.typing ArrayLike`. Which prevents it from (unhelpfully) expanding it out.

File: setup.cfg (hunk 2 above)
Comment (arcondello) [reply setup.cfg:50]: Ah, it used to be an error, but looks like they changed it to a warning in [63.4.2](https://setuptools.pypa.io/en/stable/history.html#v63-4-2).

**Pull request merged**